spec/sec-regexp-regular-expression-objects-patch.html

```diff
@@ -280,71 +280,50 @@ <h1><ins>GetStringIndex ( _S_, _Input_, _e_ )</ins></h1>
       <p>The abstract operation GetStringIndex with with arguments _S_, _Input_, and _e_ performs the following steps:</p>
       <emu-alg>
       1. Assert: Type(_S_) is String.
-      1. Assert: _Input_ is a List of Unicode code points.
-      1. _e_ is an index into the _Input_ character list, derived from _S_. Let _eUTF_ be the smallest index into _S_ that corresponds to the character at element _e_ of _Input_. If _e_ is greater than or equal to the number of elements in _Input_, then _eUTF_ is the number of code units in _S_.
+      1. Assert: _Input_ is a List of the code points of _S_ interpreted as a UTF-16 encoded string.
+      1. Assert: _e_ is an integer index within _Input_.
+      1. Let _eUTF_ be the smallest index into _S_ that corresponds to the character at element _e_ of _Input_. If _e_ is greater than or equal to the number of elements in _Input_, then _eUTF_ is the number of code units in _S_.
       1. Return _eUTF_.
       </emu-alg>
     </emu-clause>
 
     <emu-clause id="sec-getmatchresult" aoid="GetMatchResult">
-      <h1><ins>GetMatchResult ( _Input_, _startIndex_, _endIndex_, _fullUnicode_ _matchOptions_ )</ins></h1>
-      <p>The abstract operation GetMatchResult with arguments _Input_, _startIndex_, _endIndex_, _fullUnicode_, and _matchOptions_ performs the following steps:</p>
+      <h1><ins>GetMatchResult ( _S_, _startIndex_, _endIndex_, _matchOptions_ )</ins></h1>
+      <p>The abstract operation GetMatchResult with arguments _S_, _startIndex_, _endIndex_, and _matchOptions_ performs the following steps:</p>
       <emu-alg>
-      1. Assert: _Input_ is a List of either Unicode code points or code units.
+      1. Assert: Type(_S_) is String.
+      1. Assert: _startIndex_ is an integer index into _S_.
+      1. Assert: _endIndex_ is an integer index into _S_ that is greater than or equal to _startIndex_.
       1. Assert: _matchOptions_ is a MatchOptions Record.
       1. If _matchOptions_.[[Capture]] is `"indices"`, then
         1. Return CreateArrayFromList(&laquo; _startIndex_, _endIndex_ &raquo;).
       1. Else,
-        1. Let _matchChars_ be a new List whose elements are the characters of _Input_ at indices _startIndex_ (inclusive) through _endIndex_ (exclusive).
-        1. If _fullUnicode_ is *true*, then
-          1. Assert: _matchChars_ is a List of code points.
-          1. Return the String value whose code units are the UTF16Encoding of the code points of _matchChars_.
-        1. Else _fullUnicode_ is *false*,
-          1. Assert: _matchChars_ is a List of code units.
-          1. Return the String value consisting of the code units of _matchChars_.
+        1. Return the portion of _S_ between offset _startIndex_ inclusive and offset _endIndex_ exclusive.
```

Comment:
i wonder if we might want to extract part of https://tc39.es/ecma262/#sec-string.prototype.slice into an abstract operation, and then we could reuse that here? (also not a blocker)

Reply:
This is reusing existing spec text originally in RegExpBuiltinExec in the current version of the spec.

Reply:
RegExpBuiltInExec could use some refactoring anyways :-) fine if you'd prefer not to do this tho; it was just a thought.